src/backend/distributed/utils/metadata_cache.c

```diff
@@ -677,6 +722,32 @@ InvalidateDistRelationCacheCallback(Datum argument, Oid relationId)
 			cacheEntry->isValid = false;
 		}
 	}
+
+	/* now check whether we're in the middle of dropping the extension */
+
+	/* if we're not in a transaction then we'll crash while looking up oids */
+	inTransaction = IsTransactionState();
+	if (!inTransaction)
+	{
+		return;
```

Comment:
Ah, makes sense, it is called for each of the distributed tables when they are dropped.